.devcontainer/Dockerfile

```
@@ -0,0 +1,21 @@
+
+# Store tool binaries and environments in system-wide locations
+ENV UV_TOOL_BIN_DIR=/usr/local/bin
+ENV UV_TOOL_DIR=/opt/uv/tools
+
+# Install the latest versions of cmake, ninja, lit, and prek as tools
+RUN uv tool install cmake && \
+    uv tool install ninja && \
+    uv tool install prek
+
+# Downloads and installs a prebuilt LLVM/MLIR toolchain for the current OS and architecture
+RUN curl -LsSf https://github.com/munich-quantum-software/setup-mlir/releases/latest/download/setup-mlir.sh \
```

Hadolint (2.14.0)

[warning] Set the SHELL option -o pipefail before RUN with a pipe in it. If you are using /bin/sh in an alpine image or if your shell is symlinked to busybox then consider explicitly setting your SHELL to /bin/ash, or disable this check

(DL4006)